adr/adr-responses-api.md

```diff
@@ -0,0 +1,52 @@
+# ADR: Adopt Llama Stack Responses API for client-side Agent orchestration
+
+## Context
+
+Red Hat is developing an agentic capability for its AI platforms, including RHEL AI and Red Hat OpenShift AI (RHOAI). The core reasoning engine will leverage the Llama Stack. We are faced with a choice between two available APIs within Llama Stack for implementing agentic logic:
+
+1. **Llama Stack Agents API:** An older API that provides an explicit, stateful `Agent` construct on the server. The server manages the agent's state and conversation history within a `session`. While this simplifies state management for the client, this API is not aligned with the broader AI ecosystem's direction.
+2. **Llama Stack Responses API:** A newer, stateless API designed for compatibility with the OpenAI Responses API format. This API accepts a list of tools and a chat history and returns a model-generated response, which may include one or more tool calls. It does not provide a server-side `Agent` or `session` construct, requiring the client to manage the state of the interaction.
+
+The strategic direction from the Llama Stack team is to prioritize the Responses API to maximize compatibility with the vast ecosystem of tools, SDKs, and developer knowledge built around the OpenAI API standard. The older Agents API could be deprecated, heavily refactored, or receive minimal future investment, posing a significant long-term risk.
+
+The core architectural question is: How can we build a robust, developer-friendly agentic framework on a stateless API without sacrificing the usability that an explicit `Agent` construct provides?
+
+## Decision
+
+We will adopt the **Llama Stack Responses API** as the exclusive backend interface for agentic reasoning within Red Hat's AI products.
+
+To address the absence of a server-side agent object, we will **design and implement a client-side Agent construct**. This client-side agent construct will be responsible for:
```

Comment:
I think it may be best to just defer this decision making process to: https://github.com/opendatahub-io/agents/pull/2/files

Reply:
I think that the decision of the API shape to adopt is one decision, and the existence of a server-side `Agent` abstraction is an independent one. Each can go either way independently of the other, at least in principle.

Reply:
@iamemilio , the text is intended to say that we're deferring the decision process of what would replace it to 2, but I will try to tweak the wording to be more explicit.